### ISSUE TITLE
User handles are hard to tag in discussion forums

### ISSUE THREAD

**Issue by spMohanty** [crowdAI/crowdai#401]
As the user handles on the discussion forums have a " " in the middle, it might be hard for us to parse in retrospect and link the correct referenced users in the posts.

The same goes for tagging users in the discussion forum at the moment.
For ex : 
<img width="951" alt="screen shot 2017-11-07 at 11 04 40" src="https://user-images.githubusercontent.com/1581312/32488052-ac2d669e-c3ab-11e7-8ca5-efecc78736d3.png">

To reply to this user or to refer to this user, I will have to type : "@Arnaldur Bjarnason" And its going to be very hard to parse that and figure out its a username (and proactively link it to the user profile).

**Comment by scarroll32** [comment]
I agree. I think we should enforce no spaces and convert the existing records to this format.

There is a separate PR queued up which will implement optional first and last name fields, which are independent of the username.

I will try to implement the user tagging soon ... I know it's been a long time coming :-)

**Comment by scarroll32** [comment]
When it comes to regex, spaces are evil.

**Comment by scarroll32** [comment]
Actually this is now solved.
